tools/gbench/report.py

```diff
@@ -215,64 +225,112 @@ def find_test(name):
 
-            # *If* we were asked to only display aggregates,
+            # *If* we were asked to only include aggregates,
             # and if it is non-aggregate, then skip it.
-            if display_aggregates_only and 'run_type' in bn and 'run_type' in other_bench:
+            if include_aggregates_only and 'run_type' in bn and 'run_type' in other_bench:
                 assert bn['run_type'] == other_bench['run_type']
                 if bn['run_type'] != 'aggregate':
                     continue
 
-            fmt_str = "{}{:<{}s}{endc}{}{:+16.4f}{endc}{}{:+16.4f}{endc}{:14.0f}{:14.0f}{endc}{:14.0f}{:14.0f}"
-
-            def get_color(res):
-                if res > 0.05:
-                    return BC_FAIL
-                elif res > -0.07:
-                    return BC_WHITE
-                else:
-                    return BC_CYAN
-
-            tres = calculate_change(bn['real_time'], other_bench['real_time'])
-            cpures = calculate_change(bn['cpu_time'], other_bench['cpu_time'])
-            output_strs += [color_format(use_color,
-                                         fmt_str,
-                                         BC_HEADER,
-                                         bn['name'],
-                                         first_col_width,
-                                         get_color(tres),
-                                         tres,
-                                         get_color(cpures),
-                                         cpures,
-                                         bn['real_time'],
-                                         other_bench['real_time'],
-                                         bn['cpu_time'],
-                                         other_bench['cpu_time'],
-                                         endc=BC_ENDC)]
+            measurements.append({
+                'real_time': bn['real_time'],
+                'cpu_time': bn['cpu_time'],
+                'real_time_other': other_bench['real_time'],
+                'cpu_time_other': other_bench['cpu_time'],
+                'time': calculate_change(bn['real_time'], other_bench['real_time']),
+                'cpu': calculate_change(bn['cpu_time'], other_bench['cpu_time'])
+            })
 
         # After processing the whole partition, if requested, do the U test.
         if utest:
-            output_strs += print_utest(partition,
-                                       utest_alpha=utest_alpha,
-                                       first_col_width=first_col_width,
-                                       use_color=use_color)
-
-    return output_strs
-
+            have_optimal_repetitions, cpu_pvalue, time_pvalue = calc_utest(
+                extract_field(partition, 'real_time'),
+                extract_field(partition, 'cpu_time')
+            )
+            if cpu_pvalue and time_pvalue:
+                utest_results = {
+                    'stable': have_optimal_repetitions,
```

Comment:
I'm not sure `stable` is correct word here.
It should be `sufficient_observation_count`.
It *looks* like they say the same thing, but the meaning is slightly different.

Reply:
`sufficient_observation_count` seems a bit overly wordy IMO. Perhaps, `optimal_repetitions` if `stable` is not satisfying?

Newest changes are up so please have a look at them when you find it appropriate.

Reply:
`sufficient` has a very direct meaning based on https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.mannwhitneyu.html
> Use only when the number of observation in each sample is > 20

What is the definition of `optimal`, `stable` in this case?
We don't know that having less/more repetitions would have degraded the u test meaningfulness..

Reply:
> What is the definition of optimal, stable in this case?

I was following the wording of existing code (e.g. `UTEST_OPTIMAL_REPETITIONS`) so if we decide that `optimal` is not good enough changes should be done to that parts of the source code as well.

> sufficient has a very direct meaning based on https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.mannwhitneyu.html

Alright, makes sense. So `sufficient_observation_count` it is?

Reply:
@LebedevRI ping :)